src/gtest/Makefile

```diff
@@ -0,0 +1,178 @@
+CXX ?= g++
+-include ../.make-settings
+OPT=-O3 -DNDEBUG
+
+ifeq ($(DEBUG_BUILD),1)
+	CXX+=-DDEBUG_BUILD=1
+endif
+
+# Add -m32 for 32bit builds
+ifeq ($(findstring -m32,$(CFLAGS)),-m32)
+	CXX+= -m32
+endif
+
+.DEFAULT_GOAL := all
+SOURCES := $(wildcard *.cpp)
+ALL_OBJECTS := $(patsubst %.cpp,%.o,$(SOURCES))
+DEPENDS := $(patsubst %.cpp,%.d,$(SOURCES))
+
+# Hold new gtest test to a high standard
+MORE_WARNINGS=-Wold-style-cast \
+	-Wno-variadic-macros \
+	-Werror \
+	-Wpedantic \
+	-Wextra \
+	-Wall \
+	-Wcast-align \
+	-Wcast-qual \
+	-Wframe-larger-than=32768 \
+	-Wno-strict-overflow \
+	-Wsign-compare \
+	-Werror=float-equal \
+	-Werror=missing-braces \
+	-Werror=init-self \
+	-Werror=write-strings \
+	-Werror=address \
+	-Werror=array-bounds \
+	-Werror=char-subscripts \
+	-Werror=enum-compare \
+	-Werror=empty-body \
+	-Werror=main \
+	-Werror=nonnull \
+	-Werror=parentheses \
+	-Werror=return-type \
+	-Werror=sequence-point \
+	-Werror=uninitialized \
+	-Werror=volatile-register-var \
+	-Werror=ignored-qualifiers \
+	-Wno-unused-function \
+	-Wno-missing-field-initializers \
+	-Wdouble-promotion \
+	-Wformat=2
+
+# Detect compiler
+IS_GCC := $(findstring g++,$(CXX))
+
+GCC_FLAGS :=
+
+# Only add GCC-only warnings and flags
+ifeq ($(IS_GCC),g++)
+    # GCC supports these extra warnings and -fno-var-tracking-assignments
+    MORE_WARNINGS += -Wsync-nand -Wtrampolines -Werror=logical-op -Werror=aggressive-loop-optimizations
+    GCC_FLAGS += -fno-var-tracking-assignments
+else
+    # Clang: disable warnings that are errors in gtest/gmock
+    MORE_WARNINGS += -Wno-deprecated-copy
+	MORE_WARNINGS += -Wno-keyword-macro
+endif
+
+-include $(DEPENDS)
+
+ifdef COVERAGE
+	TEST_CFLAGS += -DCOVERAGE_TEST $(COVERAGE_CFLAGS)
+endif
+
+ifdef FORCE_RUN_SKIPPED_TESTS
+	TEST_CFLAGS+=-DFORCE_RUN_SKIPPED_TESTS
+endif
+
+# Track compilation flags to force rebuild when they change
+.PHONY: .flags
+.flags:
+	@echo '$(TEST_CFLAGS)' | cmp -s - $@ || echo '$(TEST_CFLAGS)' > $@
+
+# OSS Valkey library dependencies
+VALKEY_LIB := ../libvalkey.a
```

Comment:
The Valkey client is called libvalkey. We build this as valkey.a in the top level directory. I don't know if we need to rebuild it here.